routing.go

```diff
@@ -453,7 +453,7 @@ func (dht *IpfsDHT) makeProvRecord(skey cid.Cid) (*pb.Message, error) {
 // FindProviders searches until the context expires.
 func (dht *IpfsDHT) FindProviders(ctx context.Context, c cid.Cid) ([]peer.AddrInfo, error) {
 	var providers []peer.AddrInfo
-	for p := range dht.FindProvidersAsync(ctx, c, KValue) {
+	for p := range dht.FindProvidersAsync(ctx, c, dht.bucketSize) {
```

Comment:
Find Providers should also stick to just K peers. We only want to change the bucket size value, not K.